src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -223,14 +231,58 @@ internal MetricPointStatus MetricPointStatus
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public readonly HistogramBuckets GetHistogramBuckets()
         {
-            if (this.aggType != AggregationType.Histogram && this.aggType != AggregationType.HistogramSumCount)
+            if (this.aggType != AggregationType.Histogram &&
+                this.aggType != AggregationType.HistogramSumCount &&
+                this.aggType != AggregationType.HistogramMinMax &&
+                this.aggType != AggregationType.HistogramSumCountMinMax)
             {
                 this.ThrowNotSupportedMetricTypeException(nameof(this.GetHistogramBuckets));
             }
 
             return this.histogramBuckets;
         }
 
+        /// <summary>
+        /// Gets the minimum value of the histogram associated with the metric
+        /// point if present. Check by using <see cref="HasMinMax"/>.
+        /// </summary>
+        /// <remarks>
+        /// Applies to <see cref="MetricType.Histogram"/> metric type.
+        /// </remarks>
+        /// <returns>A histogram's maximum value.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public double GetHistogramMin()
+        {
+            return this.histogramBuckets.SnapshotMin;
+        }
+
+        /// <summary>
+        /// Gets the maximum value of the histogram associated with the metric
+        /// point if present. Check by using <see cref="HasMinMax"/>.
+        /// </summary>
+        /// <remarks>
+        /// Applies to <see cref="MetricType.Histogram"/> metric type.
+        /// </remarks>
+        /// <returns>A histogram's maximum value.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public double GetHistogramMax()
+        {
+            return this.histogramBuckets.SnapshotMax;
+        }
+
+        /// <summary>
+        /// Gets whether the histogram has a valid Min and Max value.
+        /// Should be used before
+        /// <see cref="GetHistogramMin"/> and <see cref="GetHistogramMax"/> to
+        /// ensure a valid value is returned.
+        /// </summary>
+        /// <returns>A minimum and maximum value exist.</returns>
+        public bool HasMinMax()
```

Comment:
Ya that's valid, we could change to `HistogramHasMinMax` maybe?
Please share any other nits I can take care of before inclusion in its initial release :)

Reply:
the initial release train just left the station 🤣 . Lets include it in next beta. (its okay to break between betas)

Reply:
`HistogramHasMinMax` works for me!

> Please share any other nits I can take care of before inclusion in its initial release :)

I was going to suggest maybe folding it all into a single method like...

`bool TryGetHistogramMinMaxValues(out double histogramMinimumValue, out double histogramMaximumValue)`

That might be faster (probably needs inlining) with less API surface but not totally sure. Would need to measure it to see feel free to try if you are interested.

Reply:
@cijothomas Hey I was just looking at this API as part of some other work. I think we should change it! Reason being, as it is right now...

```csharp
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        public double GetHistogramMin()
        {
            return this.histogramBuckets.SnapshotMin;
        }
```

...that is a NullRef just waiting to be thrown 🤣 All the other methods in here validate the type before accessing the value.

This would be much safer and have comparable perf...

```csharp
[MethodImpl(MethodImplOptions.AggressiveInlining)]
public bool TryGetHistogramMinMaxValues(out double histogramMinimumValue, out double histogramMaximumValue)
{
   if (this.aggType == AggregationType.HistogramMinMax ||
                   this.aggType == AggregationType.HistogramSumCountMinMax)
   {
      Debug.Assert(this.histogramBuckets != null, "histogramBuckets was null");

      histogramMinimumValue = this.histogramBuckets.SnapshotMin;
      histogramMaximumValue = this.histogramBuckets.SnapshotMax;
      return true;
   }

   histogramMinimumValue = 0;
   histogramMaximumValue = 0;
   return false;
}
```

Reply:
Instead of returning `true` or `false`, could we follow the pattern used for other public methods in MetricPoint such as `GetSumLong` etc. of throwing a not supported metric type exception when called for a non-applicable metric type?

Reply:
this is slightly different. Its possible to be histogram, but min/max is not there (as min max is optional).